src/actions.js

```diff
@@ -1,51 +1,37 @@
 export const FETCH = '@@redux-saga-fetch/FETCH';
 
-function createFetchAction({ method, ...fetchOptions }) {
-  return {
-    type: FETCH,
-    payload: { method, ...fetchOptions },
-    error: null
-  };
-}
+export const withJsonResponse = success => r => r.json().then(success);
 
-export function get(url, { success, fail }) {
-  return createFetchAction({
-    method: 'GET',
-    url,
-    success,
-    fail
-  });
-}
+const createFetchActionCreator = method => (
+  url,
+  { success, fail, ...fetchOptions }
+) => ({
+  type: FETCH,
+  payload: { method, url, success, fail, ...fetchOptions },
+  error: null
+});
 
-export function post(url, { body, success, fail, ...fetchOptions }) {
-  return createFetchAction({
-    method: 'POST',
-    url,
-    body,
-    success,
-    fail,
-    ...fetchOptions
-  });
-}
+const createFetchFunction = method => {
+  const func = createFetchActionCreator(method);
+  func.json = (url, { body, headers, success, ...fetchOptions }) => {
```

Comment:
`func.json` doesn't make sense for `DELETE` does it?

Reply:
yeah i thought about it, but what about some whacko non-REST endpoint that accepts a JSON body (which is not forbidden for a DELETE)? alternatively, i could restrict it to only some methods